.ci/setup.sh

```diff
@@ -20,9 +20,7 @@ if [[ $OS_NAME == "macos" ]]; then
         # Check https://github.com/actions/runner-images/tree/main/images/macos for available
         # versions of Xcode
         sudo xcode-select -s /Applications/Xcode_14.3.1.app/Contents/Developer || exit 1
-        if [[ $TASK != "mpi" ]]; then
-            brew install gcc
-        fi
+        brew install gcc
```

Comment:
I believe that `!= "mpi"` was just a shorter way to represent some other subset of jobs and that this has been carried forward unnecessarily from older versions of these build scripts.

Proposing removing this condition to simplify the scripts a bit... if the job has `COMPILER=gcc` set, we need to install `gcc`.